tests/mapping/test_mapping_script.py

```diff
@@ -27,43 +30,47 @@ def query_and_test_result(typeOfQuery, predicate, prop, g):
 # def print_first():
 #     print("###################First Query####################################")
 
-#
-# def test_generate_knowledge_graph():
-#     metadataPath = 'testMetaData.yaml'
-#     filename = "knowledgeGraph.ttl"
-#     print("Start testing:")
-#     # Generate a testMetadata file
-#     print("Generate metadata file for test:")
-#     target_data = {'experimentName': 'BA-Losert MI E-Modul 28d v. 04.08.14 Probe 4',
-#                    'software_specification': 'MTS793|MPT|DEU|1|2|,|.|:|49|1|1|A',
-#                    'operator_timestamp': '13:25:39',
-#                    'operator_date': '01.09.2014',
-#                    'tester_name': 'Kh',
-#                    'specimen_name': 'BA-Losert E-Modul 28d v. 04.08.14 Probe 4',
-#                    'remark': 'Kraftgeregelt 3,9 kN/s',
-#                    'weight': 5342.0,
-#                    'weight_unit': 'g',
-#                    'diameter': 98.6,
-#                    'length': 300.3,
-#                    'length_unit': 'mm',
-#                    'mix_file': '2014_08_05 Rezeptur_MI.xlsx'}
-#
-#     with open(metadataPath, 'w') as yamlFile:
-#         yaml.dump(target_data, yamlFile)
-#     print("Generate knowledgeGraph:")
+
+def print_first():
+    print("###################First Query####################################")
+
+
+def test_generate_knowledge_graph():
+    metadataPath = 'testMetaData.yaml'
+    filename = "EmoduleMappedExmpl.ttl"
+    print("Start testing:")
+    # Generate a testMetadata file
+    print("Generate metadata file for test:")
+    target_data = {'experimentName': 'BA-Losert MI E-Modul 28d v. 04.08.14 Probe 4',
+                   'software_specification': 'MTS793|MPT|DEU|1|2|,|.|:|49|1|1|A',
+                   'operator_timestamp': '13:25:39',
+                   'operator_date': '01.09.2014',
+                   'tester_name': 'Kh',
+                   'specimen_name': 'BA-Losert E-Modul 28d v. 04.08.14 Probe 4',
+                   'remark': 'Kraftgeregelt 3,9 kN/s',
+                   'weight': 5342.0,
+                   'weight_unit': 'g',
+                   'diameter': 98.6,
+                   'length': 300.3,
+                   'length_unit': 'mm',
+                   'mix_file': '2014_08_05 Rezeptur_MI.xlsx'}
+
+    with open(metadataPath, 'w') as yamlFile:
+        yaml.dump(target_data, yamlFile)
+    print("Generate knowledgeGraph:")
     # generate_knowledge_graph(metadataPath, filename)
 
     # g = rdflib.Graph()
-    # result = g.parse(filename, format='ttl')
+    # g.parse(filename, format='ttl')
```

Comment:
how to we check the generated knowledge graph? We need to have some tests that perform queries, see my general comment.

Reply:
The thing is that we cannot use the queries Ilias wrote as the entities have ID now. For instance, we dont have "SpecimenDiameter" anymore, instead, we have "SpecimenDiameter_UniqueID". Once the format is finalized, we can change the query so that it takes the first part of the object and then fetch the result (like using "startsWith()")